flit_core/flit_core/config.py

```diff
@@ -439,6 +449,37 @@ def _expand_requires_extra(re):
                 yield '{} ; extra == "{}"'.format(req, extra)
 
 
+def _license_files_from_globs(project_dir: Path, globs, use_defaults = False):
+    license_files = []
+    for pattern in globs:
+        if pattern.startswith("/"):
+            raise ConfigError(
+                "Invalid glob pattern for [project.license-files]: '{}'. "
+                "Pattern must not start with '/'.".format(pattern)
+            )
+        if ".." in pattern:
+            raise ConfigError(
+                "Invalid glob pattern for [project.license-files]: '{}'. "
+                "Pattern must not contain '..'".format(pattern)
+            )
+        try:
+            files = [
+                str(file.relative_to(project_dir)).replace(osp.sep, "/")
+                for file in project_dir.glob(pattern)
+                if file.is_file()
+            ]
+        except ValueError as ex:
```

Comment:
There's another check we should do on the patterns.

> Any characters or character sequences not covered by this specification are invalid. Projects MUST NOT use such values. Tools consuming this field SHOULD reject invalid values with an error.

The spec mentions alphanumerics (for which we can rely on `str.isalnum()`), plus `_-.`, globbing characters `*?[]` and the directory separator `/`. Everything else should be rejected.